protocols/request-response/src/cbor.rs

```diff
@@ -130,24 +139,46 @@ mod codec {
         where
             T: AsyncWrite + Unpin + Send,
         {
-            let data: Vec<u8> = serde_cbor::to_vec(&resp).map_err(into_io_error).unwrap();
+            let data: Vec<u8> = serde_cbor::to_vec(Vec::new(), &resp)
+                .map_err(|err| io::Error::from(CborError::Encode(err)))?;
 
             io.write_all(data.as_ref()).await?;
 
             Ok(())
         }
     }
 
-    fn into_io_error(err: serde_cbor::Error) -> io::Error {
-        if err.is_syntax() || err.is_data() {
-            return io::Error::new(io::ErrorKind::InvalidData, err);
+    impl From<CborError<Infallible>> for io::Error {
```

Comment:
Note: Implementing this over the generic involves adding trait bounds for send, sync, etc due to io::Error specifics. I felt it better to be explicit, as `TryReserveError` is used very directly in the dep for encoding only.